doc/proxy-clients.rst

```diff
@@ -296,6 +298,57 @@ And adapt your bootstrapping code to use the cache kernel::
     $response = $cacheKernel->handle($request);
     ...
 
+Cloudflare Client
+~~~~~~~~~~~~~~~~~
+
+The `Cloudflare`_ client sends HTTP requests with the ``HttpDispatcher``. Create the
+dispatcher as explained above, making sure you set the server to the Cloudflare API `https://api.cloudflare.com`, and
+pass it to the Cloudflare client::
+
+    use FOS\HttpCache\ProxyClient\Cloudflare;
+    use FOS\HttpCache\ProxyClient\HttpDispatcher;
+
+    $httpDispatcher = new HttpDispatcher(['https://api.cloudflare.com']);
+    $cloudflare = new Cloudflare($httpDispatcher);
+
+.. note::
+
+    Cloudflare does not cache HTML pages by default, this can be enabled with `custom caching with page rules`_. All
+    invalidation is done against `Cloudflare Purge API`_.
+
+You need to pass the following options to the Cloudflare client:
+
+* ``authentication_token``: User API token for authentication against Cloudflare APIs, requires `Zone.Cache` Purge permissions.
+* ``zone_identifier``: Identifier for the Cloudflare zone you want to purge the cache for.
+
+A full example could look like this::
+
+    $options = [
+        'authentication_token' => '<user-authentication-token>',
+        'zone_identifier' => '<my-zone-identifier>',
+    ];
+
+    $httpDispatcher = new HttpDispatcher(['https://api.cloudflare.com']);
+    $cloudflare = new Cloudflare($httpDispatcher, $options);
+
+.. note::
+
+    Cloudflare supports different cache purge methods depending on your account. All Cloudflare accounts support purging
+    the cache by URL and clearing all cache items. You need a `Cloudflare Enterprise`_ account to purge by cache tags.
+
+    For purging the cache by URL please see the docs `Cloudflare Purge by URL`_ for information about headers you can
+    pass to clear the cache correctly.
```

Comment:
i am a bit confused by this sentence. does the user have to configure something specifically, or do invalidation requests in a specific way? or is that more of an implementation note? if the later, i would put it in the class phpdoc of the cloudflare ProxyClient

Reply:
It contains details on how Cloudflare purges the cache by URL and which headers you need to pass to clear the cache correctly (if these are set in HTTP responses). 

I have added this to the code phpdoc, but I think this is useful to know about in the docs too since it may confuse users if the cache is not cleared as they expect. I've reworded to try to make this clearer.